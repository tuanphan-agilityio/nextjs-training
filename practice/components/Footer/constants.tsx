import FacebookIcon from '../../public/icons/facebook.svg';
import InstagramIcon from '../../public/icons/instagram.svg';
import TwitterIcon from '../../public/icons/twitter.svg';
import LinkedinIcon from '../../public/icons/linkedin.svg';

const ICONS = {
  Facebook: <FacebookIcon />,
  Instagram: <InstagramIcon />,
  Twitter: <TwitterIcon />,
  Linkedin: <LinkedinIcon />,
};

const FOOTER_CATEGORIES = [
  {
    title: 'Home',
    items: ['Product', 'Categories', 'Shop', 'Login'],
  },
  {
    title: 'Shop',
    items: ['T-Shirt', 'Jacket', 'Shirt', 'Jens'],
  },
  {
    title: 'Categories',
    items: ['Men', 'Children', 'Women'],
  },
  {
    isContact: true,
    title: 'Contact',
    items: [ICONS.Facebook, ICONS.Twitter, ICONS.Instagram, ICONS.Linkedin],
    email: 'mangcoding@gmail.com',
  },
];

export { FOOTER_CATEGORIES };
