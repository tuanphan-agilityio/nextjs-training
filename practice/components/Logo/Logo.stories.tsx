import { Story, Meta } from '@storybook/react';

import Logo from '.';

export default {
  component: Logo,
  title: 'Logo',
} as Meta;

const Template: Story = (args) => <Logo {...args} />;

export const Default = Template.bind({});
