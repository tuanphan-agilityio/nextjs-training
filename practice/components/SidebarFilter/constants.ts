const FILTER_CATEGORIES = ['Men', 'Women', 'Children'];

export { FILTER_CATEGORIES };
