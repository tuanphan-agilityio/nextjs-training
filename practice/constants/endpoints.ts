const ENDPOINTS = {
  PRODUCTS: '/products',
};

export { ENDPOINTS };
