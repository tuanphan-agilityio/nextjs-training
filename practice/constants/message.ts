const TOAST_MESSAGES = {
  UPDATED_CART: 'Cart updated successfully',
};

export { TOAST_MESSAGES };
