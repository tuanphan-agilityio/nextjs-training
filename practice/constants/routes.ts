const ROUTES = {
  HOME: '/',
  SHOP: '/shop',
  ABOUT: '/about',
  PRODUCT: '/[productId]',
  CART: '/cart',
};

export { ROUTES };
