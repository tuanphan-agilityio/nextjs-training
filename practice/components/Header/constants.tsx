import { ROUTES } from '@/constants/routes';

import CartIcon from '../../public/icons/cart.svg';
import WishlistIcon from '../../public/icons/wishlist.svg';
import UserIcon from '../../public/icons/user.svg';
import SearchIcon from '../../public/icons/search.svg';

import Input from '../Input';

const getNavItems = () => [
  {
    title: 'Home',
    href: ROUTES.HOME,
  },
  {
    title: 'Shop',
    href: ROUTES.SHOP,
  },
  {
    title: 'About Us',
    href: ROUTES.ABOUT,
  },
];

const getControlItems = () => [
  <CartIcon />,
  <WishlistIcon />,
  <UserIcon />,
  <Input
    className='max-w-[250px] h-12 rounded-lg border border-1 border-solid border-tertiary'
    placeholder='Search'
    endDecorator={<SearchIcon />}
  />,
];

export { getNavItems, getControlItems };
