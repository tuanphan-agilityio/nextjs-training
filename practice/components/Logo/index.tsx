import { FC, memo } from 'react';
import Link from 'next/link';

import { ROUTES } from '@/constants/routes';

import LogoIcon from '../../public/icons/logo.svg';

const Logo: FC = () => {
  return (
    <Link href={ROUTES.HOME}>
      <a className='flex items-center text-sm font-quaternary-bold'>
        <LogoIcon className='mr-1' /> mangcoding Store
      </a>
    </Link>
  );
};

export default memo(Logo);
