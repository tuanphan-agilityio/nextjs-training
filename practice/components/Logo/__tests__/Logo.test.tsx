import React from 'react';
import { render } from '@testing-library/react';

import Logo from '..';
import { ROUTES } from '@/constants/routes';

describe('Logo Component', () => {
  it('renders correctly', () => {
    const { container } = render(<Logo />);
    expect(container).toMatchSnapshot();
  });

  it('displays the correct text', () => {
    const { getByText } = render(<Logo />);
    const textElement = getByText('mangcoding Store');
    expect(textElement).toBeInTheDocument();
  });

  it('contains the correct link', () => {
    const { getByRole } = render(<Logo />);
    const linkElement = getByRole('link', { name: /mangcoding Store/i });
    expect(linkElement).toHaveAttribute('href', ROUTES.HOME);
  });
});
