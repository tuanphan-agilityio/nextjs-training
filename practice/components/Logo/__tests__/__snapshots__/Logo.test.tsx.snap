// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Component renders correctly 1`] = `
<div>
  <a
    aria-label="go to home page"
    class="flex items-center text-sm font-quaternary-bold"
    href="/"
  >
    <svg
      class="mr-1"
      fill="none"
      height="34"
      width="36"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M35.046 0H.214A.215.215 0 0 0 0 .215V33.72c0 .12.096.216.214.216h34.832a.214.214 0 0 0 .214-.216V.215A.214.214 0 0 0 35.046 0Z"
        fill="#0EC5D7"
      />
      <path
        d="M9.32 7.468v18.999"
        stroke="#fff"
        stroke-width="3.082"
      />
      <path
        d="M15.012 7.468v9.463"
        stroke="#fff"
        stroke-width="3.076"
      />
      <path
        d="M19.349 7.469v3.214h5.432l3.94 8.143h3.55L26.634 7.469h-7.286Z"
        fill="#fff"
      />
      <path
        d="M20.963 14.167v12.677"
        stroke="#fff"
        stroke-width="3.56"
      />
    </svg>
     
    <span
      class="sm:hidden"
    >
      mangcoding Store
    </span>
  </a>
</div>
`;
